Orange/data/pandas_compat.py

```diff
@@ -157,13 +157,56 @@ def _is_datetime(s):
         return True
     try:
         if is_object_dtype(s):
-            pd.to_datetime(s, infer_datetime_format=True)
+            # utc=True - to allow different timezones in a series object
+            pd.to_datetime(s, infer_datetime_format=True, utc=True)
             return True
     except Exception:  # pylint: disable=broad-except
         pass
     return False
 
 
+def _convert_datetime(series, var):
+    def col_type(dt):
+        """Test if is date, time or datetime"""
+        dt_nonnat = dt[~pd.isnull(dt)]  # nat == nat is False
+        if (dt_nonnat.dt.floor("d") == dt_nonnat).all():
+            # all times are 00:00:00.0 - pure date
+            return 1, 0
+        elif (dt_nonnat.dt.date == pd.Timestamp("now").date()).all():
```

Comment:
Does pandas automatically assume today's date if only time is present?

Reply:
yep. When `pd.datetime()` is called on column of times it will be transformed to datetime64 and all dates will be today's dates